core/eolearn/core/eodata_io.py

```diff
@@ -154,6 +158,45 @@ def walk_eopatch(eopatch, patch_location, features=...):
         else:
             yield ftype, fname, fs.path.combine(name_basis, fname)
 
+def merge_eopatch(eopatch_features, filesystem_features, eopatch, fs_eopatch):
+    """ Concatenate an existing eopatch in the filesystem and a new eopatch chronologically
+    """
+
+    filesystem_features = {_to_lowercase(*feature) for feature in filesystem_features}
+    eopatch_features = {_to_lowercase(*feature) for feature in eopatch_features}
+
+    intersection = filesystem_features.intersection(eopatch_features)
+    if intersection:
+        time_periods = [(fs_eopatch.timestamp[0].strftime("%Y-%m-%d"), fs_eopatch.timestamp[-1].strftime("%Y-%m-%d")),
+                        (eopatch.timestamp[0].strftime("%Y-%m-%d"), eopatch.timestamp[-1].strftime("%Y-%m-%d"))]
+
+        mask = np.isin(fs_eopatch.timestamp, list(set(fs_eopatch.timestamp).difference(set(eopatch.timestamp))))
+
+        timestamp = eopatch.timestamp + [tstamp for tstamp, to_keep in zip(fs_eopatch.timestamp, mask) if to_keep]
+        sorted_indices = sorted(range(len(timestamp)), key=lambda k: timestamp[k])
```

Comment:
just a comment, I'm usually more used to using `np.argsort`, which already gives you the indices of the sorted array, without the need for lambdas. Perhaps it's a bit clearer, but I'm not 100 % sure how it behaves with datetime objects.